src/generateTasks.js

```diff
@@ -25,15 +23,13 @@ module.exports = function generateTasks(config, relFiles) {
 
     const fileList = micromatch(
       files
-        // Only worry about children of the CWD
-        .filter(file => pathIsInside(file, cwd))
-        // Make the paths relative to CWD for filtering
-        .map(file => path.relative(cwd, file)),
+        // Make the paths relative to gitDir for filtering
+        .map(file => path.relative(gitDir, file)),
```

Comment:
This will break existing user config. User would have configured the file patterns relative to current project root as [we recommend in the readme](https://github.com/okonet/lint-staged#filtering-files).

Reply:
@okonet We should add a test for this as well.